yt_dlp/extractor/youtube.py

```diff
@@ -807,6 +807,12 @@ def _extract_video(self, renderer):
         description = self._get_text(renderer, 'descriptionSnippet')
         duration = parse_duration(self._get_text(
             renderer, 'lengthText', ('thumbnailOverlays', ..., 'thumbnailOverlayTimeStatusRenderer', 'text')))
+        if duration is None:
+            duration = parse_duration(self._search_regex(
+                r'(?i)(ago)(?!.*\1)\s+(?P<duration>[a-z0-9 ,]+?)(?:\s+[\d,]+\s+views)?(?:\s+-\s+play\s+short)?$',
```

Comment:
why `(?!.*\1)`?

Reply:
otherwise on something like the following, it starts from the first `ago`:
`some title ago by test 13 hours ago 15 seconds 207,566 views - play Short`

there might be a better way, not sure.

Reply:
gotcha. Since we don't have a restricted pattern for `duration`